cell.cpp

```diff
@@ -0,0 +1,117 @@
+#include "cell.h"
+#include <random>
+
+Cell::Cell(int newValue){
+    resetCell();
+    value = newValue;
+}
+
+void Cell::resetCell(){
+    value = 0;
+    // xCoordinate = 0;
+    // yCoordinate = 0;
+    isFilled = false;
+
+    neighbors[Direction::UP] = nullptr;
+    neighbors[Direction::DOWN] = nullptr;
+    neighbors[Direction::LEFT] = nullptr;
+    neighbors[Direction::RIGHT] = nullptr;
+}
+
+bool Cell::cellIsMovable(){
+    /*
+    * Cell is movable if have al least one neighbor with the same
+    * value or the its value = 0
+    */
+    for(const auto& pair: neighbors){
+        Cell *neighbor = pair.second;
+        if(neighbor != nullptr){
+            if(!neighbor->isFilled || neighbor->value == value){
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+int Cell::moveCellToDirection(Direction direction){
```

Comment:
ToDirection( part of name is redundant here, moveCell(Direction direction) is ok

Reply:
done